econml/selective_regularization.py

```diff
@@ -3,193 +3,98 @@
+
+    But, letting :math:`M_{X_1} = I - X_1 (X_1^\\top  X_1)^{-1}X_1^\\top`, we see that this is
+
+    .. math::
+        \\lVert (M_{X_1} y) - (M_{X_1} X_2) \\beta_2 \\rVert + f(\\beta_2)
+
+    so finding the minimizing :math:`\\beta_2` can be done by regressing :math:`M_{X_1} y` on :math:`M_{X_1} X_2` using
+    the penalized regression method incorporating :math:`f`.  Note that these are just the residual values of :math:`y`
+    and :math:`X_2` when regressed on :math:`X_1` using OLS.
     """
 
-    # TODO: allow different subsets for L1 and L2 regularization?
-
-    def __init__(self, num_outcomes, num_features, subset, steps=1000, alpha_l1=0.1, alpha_l2=0.1, learning_rate=0.1):
-        self._subset = subset
-        self._subset_c = np.setdiff1d(np.arange(num_features), subset)
-        self._steps = steps
-        self._alpha_l1 = alpha_l1
-        self._alpha_l2 = alpha_l2
-        self._learning_rate = learning_rate
-        self.tf_graph_init(num_outcomes, len(self._subset), len(self._subset_c))
-
-    def tf_graph_init(self, num_outcomes, num_reg_features, num_ureg_features):
-        """
-        Create the graph that corresponds to the squared loss with an L1/L2 penalties.
-
-        The penalties apply only on the subset of features specified by the self._subset variable.
-        Also creates the optimizer that minimizes this loss and a persistent tensorflow
-        session for the class.
-        """
-        self.Y = tf.placeholder("float", [None, num_outcomes], name="outcome")
-        self.X_reg = tf.placeholder("float", [None, num_reg_features], name="reg_features")
-        self.X_ureg = tf.placeholder("float", [None, num_ureg_features], name="ureg_features")
-
-        self.weights_reg = tf.Variable(tf.random_normal(
-            [num_reg_features, num_outcomes], 0, 0.1), name="weights_reg")
-        self.weights_ureg = tf.Variable(tf.random_normal(
-            [num_ureg_features, num_outcomes], 0, 0.1), name="weights_ureg")
-        self.bias = tf.Variable(tf.random_normal(
-            [num_outcomes], 0, 0.1), name="biases")
-        self.Y_pred = tf.add(tf.add(tf.matmul(self.X_reg, self.weights_reg),
-                                    tf.matmul(self.X_ureg, self.weights_ureg)), self.bias)
-
-        regularization = tf.contrib.layers.apply_regularization(
-            tf.contrib.layers.l1_l2_regularizer(
-                scale_l1=self._alpha_l1, scale_l2=self._alpha_l2), [self.weights_reg])
-        self.cost = tf.reduce_mean(tf.pow(self.Y - self.Y_pred, 2)) + regularization
-
-        self.optimizer = tf.train.AdagradOptimizer(learning_rate=self._learning_rate)
-        self.train = self.optimizer.minimize(self.cost)
-
-        self.session = tf.Session()
+    def __init__(self, unpenalized_inds, penalized_model, fit_intercept=False):
+        self._unpenalized_inds = unpenalized_inds
+        self._penalized_model = clone(penalized_model)
+        self._fit_intercept = fit_intercept
 
     def fit(self, X, y):
```

Comment:
Fit should support sample weights. Should be passed to all the models (both linear regressions and penalized model)

Reply:
Would require adding tests that it is doing the right thing and that running this with summarized data and weights is the same as on the unsummarized data without weights.

Reply:
Fixed (and test added)